### PR TITLE
Rewrite user profile settings (legacy routes killed, replaced by Vue.js + API)

### PR DESCRIPTION
**Pull request by RaitoBezarius**
Self-explanatory.

The user profile settings tab become:

Screenshot

Some styling / polishing must be done before merge.

### CODE REVIEW

**Comment by jilljenn** [comment]
The CDN for one static file returns 504.

File: mangaki/mangaki/static/js/profile_settings.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+          method: 'PUT',
+          headers: {
+            'Content-Type': 'application/json'
+          },
+          credentials: 'same-origin',
+          body: JSON.stringify(payload)
+        }).then(resp => {
+          return resp.json();
+        }).then(data => {
+          console.log(data)
```
Comment (jilljenn) [inline mangaki/mangaki/static/js/profile_settings.js]: Is this necessary?

File: mangaki/mangaki/static/js/profile_settings.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+      receivesNewsletter: window.INITIAL_DATA.receivesNewsletter
+    },
+    beforeUpdate: function () {
+      this.updateProfile();
+    },
+    methods: {
+      updateProfile: function () {
+        const payload = {
+          is_shared: this.isShared,
+          nsfw_ok: this.acceptsNSFW,
```
Comment (jilljenn) [inline mangaki/mangaki/static/js/profile_settings.js]: Très élégant :-)

File: mangaki/mangaki/static/js/profile_settings.js (hunk 1 above)
Comment (RaitoBezarius) [reply mangaki/mangaki/static/js/profile_settings.js]: Absolutely not :D. Debug logging for test purpose :D.

File: mangaki/mangaki/static/js/profile_settings.js (hunk 2 above)
Comment (RaitoBezarius) [reply mangaki/mangaki/static/js/profile_settings.js]: Ça le sera encore plus lorsqu'on aura formalisé une API publique et donc que les attributs ne changeront plus !

**Pull request merged**